FormalConjectures/ErdosProblems/1074.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Erdős Problem 1074
+
+*Reference:* [erdosproblems.com/1074](https://www.erdosproblems.com/1074)
+-/
+
+open scoped Nat
+
+/-- The EHS numbers (after Erdős, Hardy, and Subbarao) are those $m\geq 1$ such that there
+exists a prime $p\not\equiv 1\pmod{m}$ such that $m! + 1 \equiv 0\pmod{p}$.-/
+abbrev Nat.EHSNumbers : Set ℕ := { m | 1 ≤ m ∧ ∃ p, ¬ p ≡ 1 [MOD m] ∧ p ∣ m ! + 1}
+
+/-- The Pillai primes are those primes $p$ such that there exists an $m$ with
+$p\not\equiv 1\pmod{m}$ such that $m! + 1 \equiv 0\pmod{p}$-/
+abbrev Nat.PillaiPrimes : Set ℕ := {p | p.Prime ∧ ∃ m, ¬p ≡ 1 [MOD m] ∧ p ∣ m ! + 1}
+
+namespace Erdos1074
+
+open Nat
+
+/-- Let $S$ be the set of all $m\geq 1$ such that there exists a prime $p\not\equiv 1\pmod{m}$ such
+that $m! + 1 \equiv 0\pmod{p}$. Does
+$$
+  \lim\frac{|S\cap[1, x]|}{x}
+$$
+exist? What is it? -/
+@[category research open, AMS 11]
+theorem erdos_1074.parts.i : (∃ c, EHSNumbers.HasDensity c) ↔ answer(sorry) := by
+  sorry
```

Comment:
Arguably there should also be a version of this statement that allows the solver to provide the value of the limit using `answer(sorry)`